includes/wikia/services/WikiDetailsService.class.php

```diff
@@ -280,18 +280,29 @@ protected function cacheWikiData( $wikiInfo, $method = null ) {
 
 	/**
 	 * @param int $id
-	 * @return mixed
+	 * @return int
 	 */
 	private function getDiscussionStats( $id ) {
 		$discussionsServiceUrl = ServiceFactory::instance()->providerFactory()->urlProvider()->getUrl( 'discussion' );
-		$response = Http::get( "http://$discussionsServiceUrl/$id/forums/$id?limit=1", 'default', [ 'noProxy' => true ] );
-		if ( $response !== false ) {
-			$decodedResponse = json_decode( $response, true );
-			if ( isset( $decodedResponse['threadCount'] ) && json_last_error() === JSON_ERROR_NONE ) {
-				return $decodedResponse['threadCount'];
+		$response = Http::get( "http://$discussionsServiceUrl/$id/forums", 'default', [ 'noProxy' => true ] );
+		$threadCount = 0;
+
+		if ($response === false) {
+			return $threadCount;
```

Comment:
Shouldn't we return null here? The current behavior is that if wiki doesn't have discussions "discussions" field is not present in the response. I'm not sure but if we return 0 here would it mean that "discussions" field will be present for all wikis (even without discussions)?

Reply:
That's a good point! I'll update that.